### PR TITLE
Borer overhaul

### PR DESCRIPTION
**Pull request by N3X15**
# CHANGES

tl;dr Borers now symbiotes rather than parasites. Think organic pAIs.  Fixes #49.
- Cannot suicide as host anymore.
- Cannot kill your host anymore.
- Thought-speak is more subtle.
- Borers can secrete more chemicals (some of which are unlockable).
- Borers have a new Evolve verb to unlock new chems (and eventually, abilities and <s>host-related things</s>)
- Removed dickish chems.
- Forced-control on entering host is configurable. Add borer_takeover_immediately to gameconfig.txt to enable.
- Removed borers' ability to multiply, for now. (Balancing)
- Borers now only have a survive objective.
- Borers are instructed to be helpful and non-dickish, including a note of recent changes.
# TODO
- [x] **BYOND BUG:** Verbs don't get sent to clients of mobs inside non-turfs
- [x] Fix ventcrawling
- [ ] Verify hearing host as borer and captive_brain
- [x] Game-mode replaced with vermin event.
- [x] need to make a browse window for the Evolve verb
  - [x] abstract it into a research-tree datum for re-use in RD.

### CODE REVIEW

**Comment by PsiOmegaDelta** [comment]
Concerning that Byond bug, IIRC Bay worked around that by having an object with the relevant verbs that was given to/taken from the mob as appropriate.
May not feel like a reasonable approach but might be good to know still.

**Comment by N3X15** [comment]
@PsiOmegaDelta I'll have to talk to them about stealing it.

**Comment by PsiOmegaDelta** [comment]
This wasn't for borers specifically, we used that method when we had similar issues with AIs inserted into rigs and then given control (and didn't originally get the verbs).
The following file has the implementation: https://github.com/Baystation12/Baystation12/blob/master/code/modules/clothing/spacesuits/rig/modules/computer.dm

File: code/modules/events/infestation.dm [hunk 1]
```diff
@@ -83,6 +87,10 @@
 		if(VERM_BATS)
 			spawn_types = /mob/living/simple_animal/hostile/scarybat
 			vermstring = "bats"
+		if(VERM_BATS)
```
Comment (Zuhayr) [inline code/modules/events/infestation.dm]: Should be VERM_BORERS I assume.

**Comment by Probe1** [comment]
:+1:

File: code/modules/events/infestation.dm (hunk 1 above)
Comment (N3X15) [inline code/modules/events/infestation.dm]: Fuck, I thought I fixed that
